src/components/autocomplete/js/highlightController.js

```diff
@@ -0,0 +1,25 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.autocomplete')
+      .controller('MdHighlightCtrl', MdHighlightCtrl);
+
+  function MdHighlightCtrl ($scope, $element, $interpolate) {
+    var term = $element.attr('md-highlight-text'),
+        text = $interpolate($element.text())($scope.$parent);
```

Comment:
To be honest, I'm not 100% sure where the extra scope is coming from, but `$scope.$parent` is absolutely necessary.  As for interpolating manually, this is because I am modifying the HTML manually, and allowing Angular to control it resulted in issues.